tasks/Module 5/task_1.py

```diff
@@ -0,0 +1,125 @@
+"""Inventory Revisited."""
+
+
+import collections
+from dataclasses import dataclass
+
+MAX_INVENTORY_CAP = 80
+HEAVY_WEIGHT_THRESHOLD = 70
+LIGHT_WEIGHT_THRESHOLD = 60
+
+
+@dataclass
+class Item:
+    """Item data class."""
+
+    name: str
+    weight: int
+    price: int
+
+
+class Inventory:
+    """Inventory class."""
+
+    RUBBISH = ['rubbish', 'chewed gum', 'used tissue']
+    FOOD = ['apple', 'wine', 'chicken', 'bread']
+
+    def __init__(self):
+        """Init inventory."""
+        super().__init__()
+        self.inventory = collections.defaultdict(int)
+        self.storage = []
+
+    def add_item(self, item):
+        """Add items to inventory."""
+        if item.name in self.RUBBISH:
+            print(f'{item.name} was no added to inventory as it is rubbish\n')
+        else:
+            self.inventory[item.name] += 1
+            self.storage.append(item)
+
+    def remove_item(self, item_name):
+        """Remove item from inventiry."""
+        if self.inventory[item_name] > 0:
+            for i, item in enumerate(self.storage):
+                if item.name == item_name:
+                    del self.storage[i]
+                    print(f'Removed {item} from Inventory\n')
+                    self.inventory[item_name] -= 1
+                    break
+
+    def load_items(self, items):
+        """Load lost of items to inventory."""
+        skipped = collections.defaultdict(int)
+        added_items_number = 0
+        for item in items:
+            if item.name in self.RUBBISH:
+                skipped[item.name] += 1
+            else:
+                self.inventory[item.name] += 1
+                self.storage.append(item)
+                added_items_number += 1
+        print(f'added {added_items_number} items to inventory')
+        print('Skipped:')
+        for key, value in skipped.items():
+            print(value, key)
+        print()
+
+    def check_weight(self):
+        """Check waight and return severity."""
+        severity = 0
+        items_weight = sum(map(lambda x: x.weight, self.storage))
+        if LIGHT_WEIGHT_THRESHOLD <= items_weight < HEAVY_WEIGHT_THRESHOLD:
+            print('CAUTION: Your backpack weighs a lot, your stamina runs out quicker!')
+            severity = 1
+        elif HEAVY_WEIGHT_THRESHOLD <= items_weight < MAX_INVENTORY_CAP:
+            print('CAUTION: Your equipment is very heavy, you\'re moving slower than usual!')
+            severity = 2
+        elif items_weight >= MAX_INVENTORY_CAP:
+            print('CAUTION: You are overloaded, can\'t move!')
+            severity = 3
+        return severity
+
+    def get_weight(self):
+        """Get weight."""
+        return sum(map(lambda x: x.weight, self.storage))
```

Comment:
Two more advanced Python features, but maybe you would like them:

1. Instead of `map` (and `filter`) you can also use list comprehension or [generator expressions](https://realpython.com/introduction-to-python-generators/#building-generators-with-generator-expressions). Generator expression is especially handy, because it allows you rewrite this line as `sum(x.weight for x in self.storage)`. 
2. In Python if you have methods that just calculate and return something you can convert them to properties using `@property` decorator:
```python
@property
def weight(self):
    return sum(x.weight for x in self.storage)
```
This way you can access returned value just like field of the object: `inventory.weight` instead of `inventory.get_weight()`. Just don't use it for methods with side effects, because it is very misleading.